File: C#/Loon2MonoGame/LoonMonoGame-Lib/loon/utils/MessageQueue.cs
namespace loon.utils
{
    public class MessageQueue
    {
        public class Message
        {
            public string text;

            public int ticksRemaining;

            public Message(string text)
            {
                this.text = text;
                ticksRemaining = 100;
            }
        }

        private Array<Message> messages;
        private readonly int maxSize;

        public MessageQueue(int size)
        {
            maxSize = size;
            messages = new Array<Message>();
        }

        public Array<Message> GetArrays()
        {
            return messages.Cpy();
        }

        public void Tick()
        {
            Message mes = messages.Next();
            mes.ticksRemaining--;
            if (mes.ticksRemaining <= 0)
            {
                messages.Remove(mes);
            }
        }

        public int GetMaxSize()
        {
            return maxSize;
        }

        public int Size()
        {
            return messages.Size();
        }

        public string Get(int i)
        {
            if (i < Size())
            {
                return messages.Get(i).text;
            }
            else
            {
                return null;
            }
        }

        public void Add(string text)
        {
            messages.Add(new Message(text));
            for (; messages.Size() > maxSize;)
            {
                messages.Remove(0);
            }
        }
    }
}

File: Java/Loon-Neo/src/loon/utils/MessageQueue.java
package loon.utils;

/**
 * 一个简单的信息存储池
 *
 */
public class MessageQueue {

	public static class Message {

		public String text;

		public int ticksRemaining;

		public Message(String text) {
			this.text = text;
			ticksRemaining = 100;
		}
	}

	private Array<Message> messages;
	private final int maxSize;

	public MessageQueue(int size) {
		maxSize = size;
		messages = new Array<Message>();
	}

	public Array<Message> getArrays() {
		return messages.cpy();
	}

	public void tick() {
		Message mes = messages.next();
		mes.ticksRemaining--;
		if (mes.ticksRemaining <= 0) {
			messages.remove(mes);
		}
	}

	public int getMaxSize() {
		return maxSize;
	}

	public int size() {
		return messages.size();
	}

	public String get(int i) {
		if (i < size()) {
			return messages.get(i).text;
		} else {
			return null;
		}
	}

	public void add(String text) {
		messages.add(new Message(text));
		for (; messages.size() > maxSize;) {
			messages.remove(0);
		}
	}
}
